This folder contains reference files required to run cross-trait or GSA-MiXeR analysis, and additional files (e.g. synthetic data or selected GWAS summarys statistics) used in tutorials.

* ldsc/1000G_EUR_Phase3_plink - reference files for both cross-trait and GSA-MiXeR analyses, on the 1kG EUR template (9997231 SNPs)
* ldsc/baselineLD_v2.2_bedfiles_only_binary - functional annotations (raw format, .bed intervals) as used in GSA-MIXeR baseline model 
* sumstats - selected GWAS summary statistics for tutorials
* hapgen - synthetic genotype & sumstats data, 149454 SNPs, 10000 subjects
* mixer_hello_world - small data package to run 'hello world' examples from the documentation

Files specific to GSA-MiXeR analysis:

* gene definitions, fetched from Entrez NCBI database
* Gene Ontology gene-set definitions, fetched MSigDB database
* SynGO gene-set definitions, fetched directly from SynGO database
* Functional annotations (baselineLD_v2.2) fetched from https://alkesgroup.broadinstitute.org/LDSCORE
* The set of M=12926669 SNPs used for analyses based on UK Biobank referenceÂ 
* The set of M=11980511 SNPs used for analyses based on Haplotype Reference Consortia reference

